server/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -782,6 +783,27 @@ public final void writeOptionalInstant(@Nullable Instant instant) throws IOExcep
             o.writeInt(period.getYears());
             o.writeInt(period.getMonths());
             o.writeInt(period.getDays());
+        }),
+        entry(GenericNamedWriteable.class, (o, v) -> {
+            final var genericNamedWriteable = (GenericNamedWriteable) v;
+            final var minSupportedVersion = TransportVersion.max(
```

Comment:
This will always return the value of `genericNamedWriteable.getMinimalSupportedVersion()` since the concept of `GenericNamedWritable` was only introduced at this version.

Reply:
That's true, as long `getMinimalSupportedVersion()` is implemented correctly and not (say) copied from one of the implementations that returns `TransportVersion.MINIMUM_COMPATIBLE`. However BwC serialization is fairly subtle, and bugs in this area are just awful, so I'd like the extra check.

(We could also reasonably assert that `genericNamedWriteable.getMinimalSupportedVersion().onOrAfter(V_8_500_069)`)

Reply:
I've added the assert now